### PR TITLE
[API Compatibility] Support legacy reduction args for BCE/MSE/multi_margin/nll_loss

### PR DESCRIPTION
**Pull request by Aidenwu0209**
### PR Category

User Experience

### PR Types

New features

### Description

Complete legacy reduction argument compatibility for these functional loss APIs:

- `paddle.nn.functional.binary_cross_entropy`
- `paddle.nn.functional.mse_loss`
- `paddle.nn.functional.multi_margin_loss`
- `paddle.nn.functional.nll_loss`

This PR keeps public API signatures unchanged and handles legacy `size_average` / `reduce` through the shared legacy reduction decorator path. It also keeps PyTorch-style `target` alias support for the affected functional loss APIs.

Related issue: #76301

### 是否引起精度变化

否

### CODE REVIEW

File: python/paddle/nn/functional/loss.py [hunk 1]
```diff
@@ -46,6 +47,67 @@
 kIgnoreIndex = -100
 
 
+def _resolve_legacy_reduction_args(
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/loss.py]: 这个你修改原来的legacy_reduction_decorator、legacy_reduction_special_decorator，原本的代码上是warning，你改成直接设置size_average/reduce

File: python/paddle/nn/functional/loss.py [hunk 2]
```diff
@@ -46,6 +47,67 @@
...
+        legacy_ignore_index = ignore_index
+        ignore_index = size_average
+        size_average = None
+        if isinstance(legacy_ignore_index, str):
+            reduction = legacy_ignore_index
+
+    return size_average, ignore_index, reduce, reduction
+
+
 def dice_loss(
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/loss.py:53]: 只加装饰器，不改API签名，这两个废弃参数避免直接加到API上。

File: test/legacy_test/test_bce_loss.py [hunk 3]
```diff
@@ -299,6 +300,157 @@ def test_BCELoss_target_alias_static(self):
         )
 
 
+class TestBCELossLegacyReductionArgs(unittest.TestCase):
```
Comment (zhwesky2010) [inline test/legacy_test/test_bce_loss.py]: legacy_reduction_decorator、legacy_reduction_special_decorator涉及的十几个API都同等情况处理size_average/reduce测试，都加下

File: python/paddle/utils/decorator_utils.py [hunk 4]
```diff
@@ -726,61 +772,169 @@ def compute_legacy_reduction(reduce_val, size_average_val):
...
 ) -> Callable[_InputT, _RetT]:
-    """
-    Function decorator for __init__: intercept deprecated 'reduce' and 'size_average'.
-    """
+    sig = inspect.signature(init_func)
+    param_names = list(sig.parameters)
+    is_method = len(param_names) > 0 and param_names[0] == 'self'
 
     @functools.wraps(init_func)
     def wrapper(*args: _InputT.args, **kwargs: _InputT.kwargs) -> _RetT:
```
Comment (zhwesky2010) [inline python/paddle/utils/decorator_utils.py:852]: 不要改这么多，人工开发找到最简最优的方案写法

直接把这里的raise_deprecated_error展开，并设置kwarg['reduce']=suggested 就可以了吧？

Image

File: python/paddle/utils/decorator_utils.py [hunk 5]
```diff
@@ -789,31 +943,7 @@ def legacy_reduction_special_decorator(
...
-                    and len(use_args) > 0
-                    and use_args[0] in {'mean', 'sum', 'none', 'batchmean'}
-                )
-            ):
-                raise_deprecated_error(cls_name, reduce_val, size_avg_val)
-        return init_func(*args, **kwargs)
-
-    wrapper.__signature__ = inspect.signature(init_func)
-    return wrapper
+    return _legacy_reduction_wrapper(init_func, special=True)
```
Comment (zhwesky2010) [inline python/paddle/utils/decorator_utils.py]: special的作用是

File: python/paddle/nn/functional/loss.py [hunk 6]
```diff
@@ -1875,6 +1888,7 @@ def kl_div(
         return loss
 
 
+@legacy_reduction_decorator
 @param_one_alias(["label", "target"])
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/loss.py]: 这些别名装饰器融合到legacy_reduction_decorator里，每个API只配置一个装饰器

**Review by zhwesky2010 — COMMENTED** [review]
这个代码是人写的还是AI写的？看起来把简单的事情复杂化了。

单测你可以用AI，但是要符合单测规范：

https://github.com/PaddlePaddle/docs/blob/develop/docs/dev_guides/coding_agent/api_compatibility/python-decorator.md#step-5-%E6%B7%BB%E5%8A%A0%E6%B5%8B%E8%AF%95%E7%94%A8%E4%BE%8B

其他代码需要人先想清楚方案，并给出一定的约束规则，不是直接把需求丢进去让AI自由发挥。

这里其实可以复用之前很多逻辑，但是有大量的代码重构，有两个疑问：
1. 这里的方案思路是？
2. 选择不复用代码的原因是？
